packages/wallet/api/src/marshal-contexts.js

```diff
@@ -0,0 +1,257 @@
+// @ts-check
+import { makeScalarMap } from '@agoric/store';
+import { Far, makeMarshal, Remotable } from '@endo/marshal';
+import { HandledPromise } from '@endo/eventual-send'; // TODO: convince tsc this isn't needed
+
+const { details: X, quote: q } = assert;
+
+/**
+ * Make context for exporting wallet data where brands etc. can be recognized by boardId.
+ *
+ * Objects should be registered before serialization:
+ *   - for closely-held object, use initPurseId etc.
+ *   - for published objects such as brands and issuers, use initBoardId / ensureBoardId
+ *
+ * Unregistered objects (such as instances in amounts) get serialized
+ * with a fresh slot.
+ */
+export const makeExportContext = () => {
+  const myData = {
+    purse: {
+      /** @type {MapStore<string, Purse>} */
+      bySlot: makeScalarMap(),
+      /** @type {MapStore<Purse, string>} */
+      byVal: makeScalarMap(),
+    },
+    payment: {
+      /** @type {MapStore<string, Payment>} */
+      bySlot: makeScalarMap(),
+      /** @type {MapStore<Payment, string>} */
+      byVal: makeScalarMap(),
+    },
+    // TODO: offer, contact, dapp
+  };
+  const sharedData = {
+    /** @type {MapStore<string, unknown>} */
+    bySlot: makeScalarMap(),
+    /** @type {MapStore<unknown, string>} */
+    byVal: makeScalarMap(),
+  };
+
+  /**
+   * @param {string} slot
+   * @param {string} _iface
+   */
+  const slotToVal = (slot, _iface) => {
+    if (sharedData.bySlot.has(slot)) {
+      return sharedData.bySlot.get(slot);
+    }
+    const kind = Object.keys(myData).find(k => slot.startsWith(k));
+    assert(kind, X`bad slot kind: ${slot}`);
+    const id = slot.slice(kind.length + 1);
```

Comment:
is a slot string meant to be parseable? I don't know that as part of the commitment of the slot identifier.

Either way I'd suggest that the parsing function come from the package that composes the slot strings. For something like,

```suggestion
    const kind = Object.keys(myData).find(k => slotMatchesKind(slot, k));
    assert(kind, X`bad slot kind: ${slot}`);
    const id = idForSlotKind(slot, kind);
```

I see below that slot strings are also being composed here. If being composable and parseable by any code is a commitment of the slot design, then I'd request that the type `Slot` (LiveSlot?) be defined and its format documented on the type. I'll accept "out of scope" for addressing generally but would ask here for a simple typedef at the top with a ticket to solve generally.

Reply:
The slot type is generic. I can make this code more clear about the type we're using here, but we're not importing it.

Reply:
Ah, gotcha. It's a generic (template) parameter of `CapData`, defined by whatever the marshaller is doing. So here's a suggestion,
```
/**
  * @typedef {string} WalletSlot
  *  
  * Slot defined as `${kind}:${id}`
  */
```

Then whenever the parameter is `slot` (typed `WalletSlot`) the reader can command click to learn what the commitment is. Even better if the `valToSlot` functions have it as the return type.